File: bindings/nodejs/examples/simple_transaction/request_funds.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, Client, initLogger } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// Run with command:
// node ./dist/wallet/01_get_funds.js

// This example requests funds from the faucet
async function run() {
    initLogger();
    if (!process.env.FAUCET_URL) {
        throw new Error('.env FAUCET_URL is undefined, see .env.example');
    }
    try {
        const wallet = new Wallet({
            storagePath: './alice-database',
        });

        const account = await wallet.getAccount('Alice');

        const accountAddresses = await account.addresses();
        console.log('Account addresses:', accountAddresses);

        const faucetResponse = await new Client({}).requestFundsFromFaucet(
            process.env.FAUCET_URL,
            accountAddresses[0].address,
        );
        console.log(faucetResponse);
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());

File: bindings/python/examples/simple_transaction/request_funds.py
from iota_sdk import Wallet
from dotenv import load_dotenv
import json
import os

# This example requests funds from the faucet

load_dotenv()

FAUCET_URL = os.environ.get('FAUCET_URL', 'https://faucet.testnet.shimmer.network/api/enqueue')

wallet = Wallet('./alice-database')

account = wallet.get_account('Alice')

address = account.addresses()[0]['address']
print(address)

response = wallet.get_client().request_funds_from_faucet(FAUCET_URL, address=address)
print(response)
